src/ios/registerNativeModule.js

```diff
@@ -50,6 +59,83 @@ const addProjectToLibraries = (libraries, file) => {
 };
 
 /**
+ * Given folder, it returns an array of all header files
+ * inside it, ignoring node_modules and examples
+ */
+const getHeadersInFolder = (folder) =>
+  glob
+    .sync('**/*.h', {
+      cwd: folder,
+      nodir: true,
+      ignore: GLOB_EXCLUDE_PATTERN,
+    })
+    .map(file => path.join(folder, file));
+
+/**
+ * Given an array of headers it returns search path so Xcode can find them
+ * If there are multiple header files located accross different folders,
+ * it finds the top-most one and returns recursive version of it
+ */
+const getHeaderSearchPath = (sourceDir, headers) => {
+  const directories = union(
+    headers.map(path.dirname)
+  );
+
+  if (directories.length === 1) {
+    return `"$(SRCROOT)/${path.relative(sourceDir, directories[0])}"`;
+  }
+
+  const directory = directories.reduce((topMostDir, currentDir) => {
+    const currentFolders = currentDir.split(path.sep);
+    const topMostFolders = topMostDir.split(path.sep);
+
+    if (currentFolders.length === topMostFolders.length
+      && lastItem(currentFolders) !== lastItem(topMostFolders)) {
+      return currentFolders.slice(0, -1).join(path.sep);
```

Comment:
If files are on the same level but from different folders, e.g.:
```
src/Base/A.h
src/Layout/B.h
```
the array will be:
['users', 'grabbou', 'project', 'node_modules', 'react-module', 'src', 'base']
['users', 'grabbou', 'project', 'node_modules', 'react-module', 'src', 'layout'];

Since length is the same (same level) but folders are different, we return src. If folders are the same, there's no need to return outer folder hence that check.